constellations/gemini-user/src/class/class.js

```diff
@@ -133,46 +133,49 @@ export class ClassPage extends React.Component {
         const sessions = this.state.sessions;
         const url = "/contact?interest=" + classObj.classId;
 
-        return (
-            <div id="view-class">
-                {this.getTitle(program, semester, classObj)}
-
-                <div id="class-description">
-                    <h4 className="red">
-                        {getFullStateName(classObj.fullState)}
-                    </h4>
-                    <h4>
-                        Grades: {program.grade1} - {program.grade2}
-                    </h4>
-                    <p>{program.description}</p>
-                </div>
-
-                <div id="class-info">
-                    <Link to={url}>
-                        <button>Register</button>
-                    </Link>
-
-                    <h3>Location</h3>
-                    {this.getLocation(location)}
-                    <h3>Times</h3>
-                    {this.getTimes(classObj, sessions)}
-                    <h3>Pricing</h3>
-                    {this.getPricing(classObj)}
-                </div>
+        if (_.isEmpty(classObj))
+            return <ErrorPage classId={this.props.classId} />;
+        else
+            return (
+                <div id="view-class">
+                    {this.getTitle(program, semester, classObj)}
+
+                    <div id="class-description">
+                        <h4 className="red">
```

Comment:
instead of calling the class red, indicate what the state means. So like `fullState` or `full`